### PR TITLE
모든 city 조회 구현

### PR DESCRIPTION
**Pull request by mcodnjs**

### CODE REVIEW

File: backend/src/test/java/hanglog/trip/presentation/CityControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(CityController.class)
+@MockBean(JpaMetamodelMappingContext.class)
+@AutoConfigureRestDocs
+@Import(RestDocsConfiguration.class)
```
Comment (LJW25) [inline backend/src/test/java/hanglog/trip/presentation/CityControllerTest.java:33]: RestDocsTest 에 해당 어노테이션 포함되어있기 때문에 해당 부분 삭제해도 됩니다! (TripControllerTest에서도 삭제해야하는데 아직 못했네요)

**Review by LJW25 — APPROVED** [review]

File: backend/src/main/java/hanglog/trip/dto/response/CityResponse.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package hanglog.trip.dto.response;
+
+import hanglog.trip.domain.City;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
```
Comment (hgo641) [inline backend/src/main/java/hanglog/trip/dto/response/CityResponse.java:9]: static import?

File: backend/src/main/java/hanglog/trip/dto/response/CityResponse.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class CityResponse {
+
+    private final Long id;
+    private final String name;
+
+    public static CityResponse of(final City city) {
+        String country = city.getCountry();
+        String name = city.getName();
```
Comment (hgo641) [inline backend/src/main/java/hanglog/trip/dto/response/CityResponse.java]: final?

File: backend/src/test/java/hanglog/trip/service/CityServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Mock
+    private CityRepository cityRepository;
+
+    @DisplayName("모든 도시를 반환한다.")
+    @Test
+    void getAllCities() {
+        // given
+        final City seoul = new City(1L, "서울", "대한민국", new BigDecimal("126.9779692"), new BigDecimal("37.566535"));
+        final City busan = new City(2L, "부산", "대한민국", new BigDecimal("129.0756416"), new BigDecimal("35.1795543"));
```
Comment (waterricecake) [inline backend/src/test/java/hanglog/trip/service/CityServiceTest.java:35]: fixture에 넣거나 fixture의 city를 쓰는건 어떤가요?

**Review by hgo641 — APPROVED** [review]
고생고생하셨습니다

File: backend/src/main/java/hanglog/trip/dto/response/CityResponse.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package hanglog.trip.dto.response;
+
+import hanglog.trip.domain.City;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
```
Comment (waterricecake) [inline backend/src/main/java/hanglog/trip/dto/response/CityResponse.java:9]: import static 어떤가요?

File: backend/src/main/java/hanglog/trip/presentation/CityController.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/cities")
+public class CityController {
+
+    private final CityService cityService;
+
+    @GetMapping
+    public ResponseEntity<List<CityResponse>> getCities() {
+        return ResponseEntity.ok(cityService.getAllCities());
```
Comment (waterricecake) [inline backend/src/main/java/hanglog/trip/presentation/CityController.java:22]:
```suggestion
        final List<CityResponse> cityResponses = cityService.getAllCities();
        return ResponseEntity.ok(cityResponses);
```
다른 컨트롤러들은 분리해두었길래....

**Review by waterricecake — COMMENTED** [review]
수고하셨습니다 라온

File: backend/src/test/java/hanglog/trip/presentation/CityControllerTest.java (hunk 1 above)
Comment (mcodnjs) [reply backend/src/test/java/hanglog/trip/presentation/CityControllerTest.java]: [refactor: RestDocsTest 어노테이션 삭제](https://github.com/woowacourse-teams/2023-hang-log/pull/105/commits/2f15d47d509fc6e8b1b030d38bcf93bfbbcb2de8)

File: backend/src/test/java/hanglog/trip/service/CityServiceTest.java (hunk 4 above)
Comment (mcodnjs) [reply backend/src/test/java/hanglog/trip/service/CityServiceTest.java]: 오 굿

Comment (mcodnjs) [reply backend/src/test/java/hanglog/trip/service/CityServiceTest.java]: [refactor: 목업데이터 fixture로 변경](https://github.com/woowacourse-teams/2023-hang-log/pull/105/commits/cec65dc09c3071d0aa723eac7e5579fc02a1daef)

File: backend/src/main/java/hanglog/trip/presentation/CityController.java (hunk 6 above)
Comment (mcodnjs) [reply backend/src/main/java/hanglog/trip/presentation/CityController.java]: [refactor: 서비스 변수 추출](https://github.com/woowacourse-teams/2023-hang-log/pull/105/commits/ce555d1db7eb4ba1ba92fce5f08dcb57a8137f2d)

File: backend/src/main/java/hanglog/trip/dto/response/CityResponse.java (hunk 2 above)
Comment (mcodnjs) [reply backend/src/main/java/hanglog/trip/dto/response/CityResponse.java]: 추가완료 !
[refactor: static import 추가](https://github.com/woowacourse-teams/2023-hang-log/pull/105/commits/0e379e6a4ffe4a31020ba5b2ed83183d2244892a)

File: backend/src/main/java/hanglog/trip/dto/response/CityResponse.java (hunk 5 above)
Comment (mcodnjs) [reply backend/src/main/java/hanglog/trip/dto/response/CityResponse.java]: [refactor: static import 추가](https://github.com/woowacourse-teams/2023-hang-log/pull/105/commits/0e379e6a4ffe4a31020ba5b2ed83183d2244892a)

**Pull request merged**